testinfra/utils/ansible_runner.py

```diff
@@ -18,150 +18,172 @@
-        play = ansible.playbook.play.Play().load({
-            "hosts": host,
-            "gather_facts": "no",
-            "tasks": [{
-                "action": action,
-            }],
-        }, variable_manager=self.variable_manager, loader=self.loader)
-        tqm = None
-        callback = Callback()
-        try:
-            tqm = ansible.executor.task_queue_manager.TaskQueueManager(
-                inventory=self.inventory,
-                variable_manager=self.variable_manager,
-                loader=self.loader,
-                options=self.cli.options,
-                passwords=None,
-                stdout_callback=callback,
+        '''Invokes a single module on a single host and returns dict results'''
+
+        # runner must have a directory based payload
+        with tempfile.TemporaryDirectory(prefix='runner.data.') as data_dir:
+
+            # runner must have an inventory file
+            inv_dir = os.path.join(data_dir, 'inventory')
+            os.makedirs(inv_dir)
+            shutil.copy(
+                self.host_list,
+                os.path.join(inv_dir, os.path.basename(self.host_list))
             )
-            tqm.run(play)
-        finally:
-            if tqm is not None:
-                tqm.cleanup()
 
-        return callback.result
+            # molecule inventories use lookups
+            this_env = os.environ.copy()
+            this_env['ANSIBLE_NOCOLOR'] = "1"
+            env_keys = list(this_env.keys())
+            for ekey in env_keys:
+                if not ekey.startswith('MOLECULE'):
+                    this_env.pop(ekey)
+            env_dir = os.path.join(data_dir, 'env')
+            if not os.path.exists(env_dir):
+                os.makedirs(env_dir)
+            with open(os.path.join(env_dir, 'envvars'), 'w') as f:
+                f.write('---\n')
+                f.write(yaml.dump(this_env))
+
+            # build the kwarg payload ansible-runner requires
+            runner_kwargs = {
+                'private_data_dir': data_dir,
+                'host_pattern': host,
+                'module': module_name,
+                'module_args': module_args,
+                'json_mode': True,
+            }
+
+            # ansible-runner does not have kwargs for these
+            for opt in ['become', 'check']:
+                if kwargs.get(opt, False):
+                    if 'cmdline' not in runner_kwargs:
+                        runner_kwargs['cmdline'] = '--%s' % opt
+                    else:
+                        runner_kwargs['cmdline'] += ' --%s' % opt
+
+            return AnsibleRunnerV2.call_runner(runner_kwargs, host)
+
+    @staticmethod
+    def call_runner(runner_kwargs, host):
+        # invoke ansible-runer -> ansible adhoc
+        r = ansible_runner.run(**runner_kwargs)
+        events = r.host_events(host)
+
+        # a "significant" event is the event that has the task result
+        significant_event = None
+        if events:
```

Comment:
This check is unnecessary: iterating through an empty iterator will just skip the loop automatically.
```suggestion
```